### PR TITLE
Add support for address information on `Customer`

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @ob-stripe 
cc @stripe/api-libraries @robertcoe-stripe

### CODE REVIEW

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 1]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
         }
         #endregion
 
+        /// <summary>
+        /// The customer’s address. Until the invoice is finalized, this field will equal
+        /// <code>customer.address</code>. Once the invoice is finalized, this field will no longer
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]:
```suggestion
        /// <c>customer.address</c>. Once the invoice is finalized, this field will no longer
```

`<code>` is for multiple lines of code, cf. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/code-inline.

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 2]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
...
+        /// be updated.
+        /// </summary>
+        [JsonProperty("customer_address")]
+        public Address CustomerAddress { get; set; }
+
+        /// <summary>
+        /// The customer’s name. Until the invoice is finalized, this field will equal
+        /// <code>customer.name</code>. Once the invoice is finalized, this field will no longer be
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]:
```suggestion
        /// <c>customer.name</c>. Once the invoice is finalized, this field will no longer be
```

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 3]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
...
+        /// </summary>
+        [JsonProperty("customer_name")]
+        public string CustomerName { get; set; }
+
+        /// <summary>
+        /// The customer’s phone number. Until the invoice is finalized, this field will equal
+        /// <code>customer.phone</code>. Once the invoice is finalized, this field will no longer
+        /// be updated.
+        /// </summary>
+        [JsonProperty("phone")]
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]:
```suggestion
        [JsonProperty("customer_phone")]
```

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 4]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
...
+        /// The customer’s name. Until the invoice is finalized, this field will equal
+        /// <code>customer.name</code>. Once the invoice is finalized, this field will no longer be
+        /// updated.
+        /// </summary>
+        [JsonProperty("customer_name")]
+        public string CustomerName { get; set; }
+
+        /// <summary>
+        /// The customer’s phone number. Until the invoice is finalized, this field will equal
+        /// <code>customer.phone</code>. Once the invoice is finalized, this field will no longer
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]:
```suggestion
        /// <c>customer.phone</c>. Once the invoice is finalized, this field will no longer
```

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 5]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
...
+        /// be updated.
+        /// </summary>
+        [JsonProperty("phone")]
+        public string CustomerPhone { get; set; }
+
+        /// <summary>
+        /// The customer’s shipping information. Until the invoice is finalized, this field will
+        /// equal <code>customer.shipping</code>. Once the invoice is finalized, this field will no
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]:
```suggestion
        /// equal <c>customer.shipping</c>. Once the invoice is finalized, this field will no
```

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 6]
```diff
@@ -103,6 +103,44 @@ internal object InternalCustomer
...
+        /// longer be updated.
+        /// </summary>
+        [JsonProperty("customer_shipping")]
+        public Shipping CustomerShipping { get; set; }
+
+        /// <summary>
+        /// The customer’s preferred locales (languages), ordered by preference.
+        /// </summary>
+        [JsonProperty("preferred_locales")]
+        public List<string> PreferredLocales { get; set; }
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]: I don't think this attribute actually exists.

**Comment by remi-stripe** [comment]
@ob-stripe PTAL (nice catch)

File: src/Stripe.net/Entities/Invoices/Invoice.cs (hunk 3 above)
Comment (ob-stripe) [reply src/Stripe.net/Entities/Invoices/Invoice.cs]: @remi-stripe Looks like you haven't fixed this one.

**Comment by remi-stripe** [comment]
ugh sorry, fixed for real this time.

File: src/Stripe.net/Entities/Invoices/Invoice.cs [hunk 7]
```diff
@@ -13,6 +13,20 @@ public class Invoice : StripeEntity, IHasId, IHasMetadata, IHasObject
...
+        [JsonProperty("account_country")]
+        public string AccountCountry { get; set; }
+
+        /// <summary>
+        /// The public name of the business associated with this invoice, most often the business
+        /// creating the invoice
+        /// </summary>
+        [JsonProperty("account_name")]
+        public string AccountName { get; set; }
+
```
Comment (remi-stripe) [inline src/Stripe.net/Entities/Invoices/Invoice.cs]: Added those two since I just realized they are new too.

**Comment by remi-stripe** [comment]
@ob-stripe Added 2 new fields can you re-approve?

**Pull request merged**